scipy/stats/stats.py

```diff
@@ -3692,6 +3693,43 @@ def f_oneway(*args, axis=0):
     return F_onewayResult(f, prob)
 
 
+def _pearsonr_fishers_ci(r, n, confidence_level, alternative):
+    """
+    Compute the confidence interval for Pearson's R.
+
+    Fisher's transformation is used to compute the confidence interval
+    (https://en.wikipedia.org/wiki/Fisher_transformation).
+    """
+    if r == 1:
+        zr = np.inf
+    elif r == -1:
+        zr = -np.inf
+    else:
+        zr = np.arctanh(r)
+    v = 1 / (n - 3)
+    s = np.sqrt(v)
+
+    if alternative == "two-sided":
```

Comment:
Makes sense. Looks good.

Reply:
But maybe add to the notes how it's calculated? Probably should mention somewhere that it's an asymptotic approximation based on certain assumptions.